plugins/poll/app/controllers/polls_controller.rb

```diff
@@ -61,18 +61,25 @@ def grouped_poll_results
     poll_name = params.require(:poll_name)
     user_field_name = params.require(:user_field_name)
 
-    begin
+    if Poll.find_by(post_id: post_id, name: poll_name).type == "irv"
       render json: {
-               grouped_results:
-                 DiscoursePoll::Poll.grouped_poll_results(
-                   current_user,
-                   post_id,
-                   poll_name,
-                   user_field_name,
-                 ),
-             }
-    rescue DiscoursePoll::Error => e
-      render_json_error e.message
+               error: "Invalid poll type, IRV does not support grouped results.",
```

Comment:
This string should be translatable.

Reply:
Done: https://github.com/discourse/discourse/pull/27155/commits/0a2f5ea5fa0153efe4e9b262e9ee11417b124edf